.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,33 @@
+name: dotnet
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 2.1.401
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build --configuration Release --no-restore
```

Comment:
are those necessary? `dotnet test` should restore and build first, shouldn't it?

Reply:
I think separating it makes it easier to figure out where the problem is if something bombs.